docs/rules.md

```diff
@@ -0,0 +1,265 @@
+# Rules
+
+Rule is the basic unit of input passed to the Rules Engine. The engine parses rules, evaluates them and generates _Violations_ when rules are matched. A collection of one or more rules form a [_Ruleset_](#ruleset). _Rulesets_ provide an opionated way of organizing multiple rules that achieve a common goal. 
+
+## Rule
+
+A Rule consists of conditions, actions and metadata. It instructs analyzer to take specified actions when given conditions match.
+
+Metadata fields of a rule are:
+
+```yaml
+# id must be unique among a Ruleset
+ruleId: "unique_id"
+# violations have pre-defined categories
+category: "potential|information|mandatory"
+# links point to external hyperlinks
+# rule authors are expected to provide
+# relevant hyperlinks for quick fixes, docs etc
+links:
+  - url: "konveyor.io"
+    title: "title to be shown in report"
+# labels provide metadata about the rule
+# can be used to find / filter rules
+labels:
+  - "label1"
+  - "label2"
+# effort is an integer value to indicate level of 
+# effort needed to fix this issue
+effort: 1
+```
+
+Actions of a rule are:
+
+```yaml
+# when a match is found, analyzer generates a violation with this message
+message: "helpful message about the violation"
+# when a match is found, analyzer generates these tags for the application
+tag:
+  - "tag1"
+  - "tag2"
+
+```
+
+Finally, a rule contains a `when` block to specify a condition:
+
+```yaml
+when:
+  <conditions>
+```
+
+### When
+
+`When` consists of a condition.
+
+#### Provider Conditions
```

Comment:
Do we want to add the provider settings docs here as well?

Reply:
@shawn I'll address that in a follow up